app/src/main/java/com/kotlin/readyplayerme/WebViewActivity.kt

```diff
@@ -126,29 +125,47 @@ class WebViewActivity : AppCompatActivity() {
     private fun handleAvatarCreated() {
         with(binding.webview){
             evaluateJavascript("""
-                window.addEventListener("message", receiveMessage, false)
-                function receiveMessage(event){                        
-                    // url display, used for displaying any string that starts with "https:"                    
-                    if (typeof data === "string" && event.data.indexOf("https:") !== -1) {
+                function subscribe(event) {
+                    // post message v1, this will be deprecated
+                    if(event.data.endsWith('.glb')) {
                         var content = document.querySelector(".content")
                         content.remove()
-                        MyScript.openDialog(event.data)
+                        WebView.receiveData(event.data)
                     }
-                    else{
-                        // catches only RPM related messages, displays stickers object
-                        // parse incoming message into JSON, and if object has readyPlayerMe and sticker
-                        // fields take stickers and print it as a string
-                          
-                        const data = JSON.parse(event.data)
+                    // post message v2
+                    else {
+                        const json = parse(event);
+                        const source = json.source;
                         
-                        if(data.readyPlayerMe && data.readyPlayerMe.stickers)
-                        {
-                            var content = document.querySelector(".content")
-                            content.remove()
-                            MyScript.openDialog(JSON.stringify(data.readyPlayerMe.stickers))
+                        if (source !== 'readyplayerme') {
+                          return;
                         }
+    
+                        var content = document.querySelector(".content")
```

Comment:
Can simplify all of these instances to
```javascript
document.querySelector(".content").remove();
```

Reply:
Btw I don't really understand the intent here in the original code. Why we modify the DOM in WebView instead of terminating the WebView itself 🤔

Reply:
Dunno but this was done by a freelancer, probs they didn't know what they were doing either